### PR TITLE
NodeConditionPredicates should return NodeOutOfDisk error.

### LINKED ISSUE #50594: NodeConditionPredicates in DaemonSet should return OutOfDisk error.
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**Description**:

In DaemonSet, `NodeConditionPredicates` should return `NodeOutOfDisk` after moved node condition check into a predicates.

### PR DESCRIPTION
**Pull request by k82cn**
**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/pull/49932 , I moved node condition check into a predicates; but it return incorrect error :(. 

We also need to add more cases to `TestNodeShouldRunDaemonPod` which is key function of DaemonSet.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #50594 

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/controller/daemon/daemon_controller_test.go [hunk 1]
```diff
@@ -1422,6 +1423,23 @@ func TestNodeShouldRunDaemonPod(t *testing.T) {
...
+					},
+				},
+			},
+			nodeCondition:         []v1.NodeCondition{{Type: v1.NodeOutOfDisk, Status: v1.ConditionTrue}},
+			wantToRun:             true,
+			shouldSchedule:        false,
+			shouldContinueRunning: true,
+		},
+		{
+			ds: &extensions.DaemonSet{
```
Comment (zhangxiaoyu-zidif) [inline pkg/controller/daemon/daemon_controller_test.go]: This is func func TestNodeShouldRunDaemonPod(t *testing.T) {(t *testing.T) 
Should its name be changed?
The test case is TestNodeShouldNotRunDaemonPod(t *testing.T)

**Comment by k82cn** [comment]
ping for review :).

File: pkg/controller/daemon/daemon_controller_test.go (hunk 1 above)
Comment (k82cn) [inline pkg/controller/daemon/daemon_controller_test.go]: nop; we're testing `nodeShouldRunDaemonPod` in this test :)

**Comment by k82cn** [comment]
@kubernetes/sig-apps-pr-reviews

File: pkg/controller/daemon/daemon_controller.go [hunk 2]
```diff
@@ -1351,7 +1351,7 @@ func NodeConditionPredicates(nodeInfo *schedulercache.NodeInfo) (bool, []algorit
 		// TODO: There are other node status that the DaemonSet should ideally respect too,
 		//       e.g. MemoryPressure, and DiskPressure
 		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
-			reasons = append(reasons, predicates.ErrNodeSelectorNotMatch)
+			reasons = append(reasons, predicates.ErrNodeOutOfDisk)
```
Comment (janetkuo) [inline pkg/controller/daemon/daemon_controller.go]: Would you add a unit test to make sure the right reasons are returned in DaemonSet predicates? Something similar to:
https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/plugin/pkg/scheduler/algorithm/predicates/predicates_test.go#L1951

Comment (k82cn) [inline pkg/controller/daemon/daemon_controller.go]: sure; that makes sense :).

Comment (janetkuo) [inline pkg/controller/daemon/daemon_controller.go]: This PR looks good otherwise

Comment (k82cn) [reply pkg/controller/daemon/daemon_controller.go]: Done; I only add one case for this PR; I'll open other PR to increase the coverage.

**Review by janetkuo — APPROVED** [review]

**Pull request merged**